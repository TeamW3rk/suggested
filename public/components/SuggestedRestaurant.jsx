import React from 'react';
import ReactDOM from 'react-dom';
import $ from 'jquery';

const SuggestedRestaurant = (props) => (
  <div id="suggested-restaurant">
    {/* <div id="suggested-restaurant-id">{props.restaurant.id}</div> */}
    <div><img src={props.restaurant.image} id="suggested-restaurant-image" /></div> 
    <div id="suggested-restaurant-name">{props.restaurant.name}</div>
    <div id="suggested-restaurant-stars">{props.restaurant.stars}</div>
    <div id="suggested-restaurant-price">{props.restaurant.price}</div>
    <div id="suggested-restaurant-type">{props.restaurant.type}</div>     
    <div id="suggested-restaurant-amount-booked">{props.restaurant.amountBooked}</div>
  </div>
)

export default SuggestedRestaurant;

