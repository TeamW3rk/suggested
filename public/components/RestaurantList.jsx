import React from 'react';
import ReactDOM from 'react-dom';
import $ from 'jquery';
import SuggestedRestaurant from './SuggestedRestaurant.jsx';

const RestaurantList = (props) => (
  <div id="restaurants">
    <div id="header">Need to book at 7:00 PM? Other great restaurants with availability.
    <button></button>
    <button></button>
    </div>
    {props.list[0] ? props.list[0].suggestedRestaurants.map((restaurant)=> {
      return <SuggestedRestaurant restaurant={restaurant} />
    }) : ""}
  </div>
)

export default RestaurantList;
