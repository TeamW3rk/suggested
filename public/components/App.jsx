import React from 'react';
import ReactDOM from 'react-dom';
import $ from 'jquery';
import RestaurantList from './RestaurantList.jsx';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      restaurants: []
    }
  }

  componentWillMount() {
    this.fetch(window.location.pathname);
  }

  fetch(path) {
    $.ajax({
      method: 'GET',
      url: 'http://localhost:3000' + path + 'suggestions',
      success: (data) => {
        console.log('SUCCESS! ', data);
        this.setState({restaurants: data})
      },
      error: (err) => {
        console.log('ERROR', err)
      }
    })
  }

  handleScroll(button) {
    if (button === '#left-scroll') {
      $(button).click(() => {
        var leftPosition = $("#restaurants").scrollLeft();
        console.log('left ', leftPosition)
        $("#restaurants").animate({scrollLeft: leftPosition - 400}, 100);
      })
    } else {
      $(button).click(() => {
        var rightPosition = $("#restaurants").scrollLeft();
        console.log('right ', rightPosition);
        $("#restaurants").animate({scrollLeft: rightPosition + 400}, 100);
      })
    }
  }

  getRestaurantId(id) {
    console.log(id);
    this.fetch('/r/' + id + '/');
  }

  render() {
    return (
    <div>
      <RestaurantList list={this.state.restaurants} scroll={this.handleScroll.bind(this)} 
                      getId={this.getRestaurantId.bind(this)} />
    </div>
    )
  }
}

export default App;
