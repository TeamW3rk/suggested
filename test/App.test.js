import React from 'react';
import App from '../public/components/App.jsx';
import RestaurantList from '../public/components/RestaurantList.jsx';
import SuggestedRestaurant from '../public/components/SuggestedRestaurant.jsx';

describe('App component', () => {
  test('RestaurantList should exist', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find(RestaurantList).length).toEqual(1);
  });
})

describe('RestaurantList component', () => {
  test('renders RestaurantList component', () => {
    const fakeList = [];
    const fakeFunc1 = jest.fn();
    const fakeFunc2 = jest.fn();
    const wrapper = shallow(<RestaurantList list={fakeList} scroll={fakeFunc1} getId={fakeFunc2}/>);
    expect(wrapper).toMatchSnapshot();
  });
})

describe('SuggestedRestaurant component', () => {
  test('renders SuggestedRestaurant component', () => {
    const fakeRestaurant = {};
    const fakeFunc1 = jest.fn();
    const wrapper = shallow(<SuggestedRestaurant restaurant={fakeRestaurant} getId={fakeFunc1} />);
    expect(wrapper).toMatchSnapshot();
  });
})




