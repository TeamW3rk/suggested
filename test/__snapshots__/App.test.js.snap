// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantList component renders RestaurantList component 1`] = `
<div>
  <div
    id="header"
  >
    Need to book at 7:00 PM? Other great restaurants with availability.
    <div
      id="buttons"
    >
      <button
        id="left-scroll"
        onClick={[Function]}
      >
        <img
          id="left-arrow"
          src="./images/left-arrow.png"
        />
      </button>
      <button
        id="right-scroll"
        onClick={[Function]}
      >
        <img
          id="right-arrow"
          src="./images/right-arrow.png"
        />
      </button>
    </div>
  </div>
  <div
    id="restaurants"
  />
</div>
`;

exports[`SuggestedRestaurant component renders SuggestedRestaurant component 1`] = `
<div
  id="suggested-restaurant"
  onClick={[Function]}
>
  <div>
    <img
      id="suggested-restaurant-image"
    />
  </div>
  <div
    id="suggested-restaurant-info"
  >
    <div
      id="suggested-restaurant-name"
    />
    <div
      id="suggested-restaurant-stars"
    >
      <span
        className="fa fa-star checked"
      />
      <span
        className="fa fa-star checked"
      />
      <span
        className="fa fa-star checked"
      />
      <span
        className="fa fa-star checked"
      />
      <span
        className="fa fa-star checked"
      />
      <span
        id="stars-number"
      >
         
         
      </span>
    </div>
    <div
      id="suggested-restaurant-price"
    >
      <span
        className="dollar"
      >
        $
      </span>
      <span
        className="dollar"
      >
        $
      </span>
      <span
        className="dollar"
      >
        $
      </span>
      <span
        className="dollar"
      >
        $
      </span>
      <span
        className="dollar"
      >
        $
      </span>
    </div>
    <div
      id="type-location"
    >
      <div
        id="suggested-restaurant-type"
      >
         
        <span
          id="middot"
        >
          Â·
        </span>
         
      </div>
      <div
        id="suggested-restaurant-location"
      />
    </div>
    <div
      id="suggested-restaurant-amount-booked"
    >
      <img
        id="booked-image"
        src="./images/booked.png"
      />
      Booked 
       times today
    </div>
  </div>
</div>
`;
